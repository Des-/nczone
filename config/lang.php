<?php

namespace eru\nczone\config;

final class lang
{
    public static function en(string $path)
    {
        return self::get('en', $path);
    }

    public static function de(string $path)
    {
        return self::get('de', $path);
    }

    private static function get(string $lang, string $path)
    {
        // todo: add deep path if that is needed
        return self::all()[$lang][$path] ?? [];
    }

    private static function all()
    {
        static $all;
        if ($all === null) {
            $all = \json_decode(\file_get_contents(__DIR__ .'/lang.json'), true);
        }
        return $all;
    }
}
