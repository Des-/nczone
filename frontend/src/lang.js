export default {
  'de': {
    // civs
    'NCZONE_AZTECS': 'Azteken',
    'NCZONE_BERBERS': 'Berber',
    'NCZONE_BRITONS': 'Briten',
    'NCZONE_BURMESE': 'Birmanen',
    'NCZONE_BYZANTINES': 'Byzantiner',
    'NCZONE_CELTS': 'Kelten',
    'NCZONE_CHINESE': 'Chinesen',
    'NCZONE_ETHOPIANS': 'Äthiopier',
    'NCZONE_FRANKS': 'Franken',
    'NCZONE_GOTHS': 'Goten',
    'NCZONE_HUNS': 'Hunnen',
    'NCZONE_INCAS': 'Inka',
    'NCZONE_INDIANS': 'Inder',
    'NCZONE_ITALIANS': 'Italiener',
    'NCZONE_JAPANESE': 'Japaner',
    'NCZONE_KHMER': 'Khmer',
    'NCZONE_KOREANS': 'Koreaner',
    'NCZONE_MAGYARS': 'Ungarn',
    'NCZONE_MALAY': 'Malaien',
    'NCZONE_MALIANS': 'Malier',
    'NCZONE_MAYANS': 'Maya',
    'NCZONE_MONGOLS': 'Mongolen',
    'NCZONE_PERSIANS': 'Perser',
    'NCZONE_PORTUGUESE': 'Portugiesen',
    'NCZONE_SARACENS': 'Sarazenen',
    'NCZONE_SLAVS': 'Slaven',
    'NCZONE_SPANISH': 'Spanier',
    'NCZONE_TEUTONS': 'Teutonen',
    'NCZONE_TURKS': 'Türken',
    'NCZONE_VIETNAMESE': 'Vietnamesen',
    'NCZONE_VIKINGS': 'Wikinger',

    'NCZONE_MATCH_RESULT': 'Ergebnis',
    'NCZONE_MATCH_POST_RESULT': 'Wähle ein Ergebnis',
    'NCZONE_MATCH_POST_WIN_TEAM1': 'Team 1',
    'NCZONE_MATCH_POST_WIN_TEAM2': 'Team 2',
    'NCZONE_MATCH_POST_OW': 'Ohne Wertung',
    'NCZONE_MATCH_SEND_RESULT': 'Ergebnis übermitteln',

    'NCZONE': 'nC Zone',
    'NCZONE_OK': 'OK',
    'NCZONE_CANCEL': 'Abbrechen',

    'NCZONE_LOGGEDIN': 'Eingeloggt',
    'NCZONE_LOGIN': 'Einloggen',
    'NCZONE_LOGOUT': 'Ausloggen',
    'NCZONE_DRAW': 'Losen',
    'NCZONE_NO_LOGIN': 'Keiner eingeloggt :(',

    'NCZONE_LINKS': 'Links',
    'NCZONE_ZONE': 'Teh Zone',
    'NCZONE_INFORMATION': 'Max wichtige Informationen',

    'NCZONE_RMATCHES': 'Laufende Spiele',
    'NCZONE_PMATCHES': 'Vergangene Spiele',
    'NCZONE_PLAYERS_TABLE': 'Tabelle',
    'NCZONE_MENU_INFORMATION': 'Informationen',
    'NCZONE_RULES': 'Regeln',
    'NCZONE_SETTINGS': 'Einstellungen',
    'NCZONE_SWITCH_LANG': 'Switch to English',

    'NCZONE_MATCH_MATCH': 'Spiel',
    'NCZONE_MATCH_DRAWER': 'Gelost von',
    'NCZONE_MATCH_RESULT_POSTER': 'Gepostet von',
    'NCZONE_MATCH_WINNER': 'Gewinner',
    'NCZONE_MATCH_WINNER_NO_RESULT': 'Ohne Wertung',
    'NCZONE_MATCH_TEAM': 'Team',
    'NCZONE_MATCH_MAP': 'Karte',
    'NCZONE_MATCH_CIVS': 'Civs',
    'NCZONE_MATCH_CIVS_BOTH': 'Beide',
    'NCZONE_MATCH_CIVS_TEAM1': 'Team 1',
    'NCZONE_MATCH_CIVS_TEAM2': 'Team 2',
    'NCZONE_MATCH_START_TIME': 'Spielstart',
    'NCZONE_MATCH_TIME_SINCE_DRAW': 'Zeit seit dem Losen',
    'NCZONE_MATCH_LENGTH': 'Länge',
    'NCZONE_MATCH_TEAMS': 'Teams',
    'NCZONE_MATCH_TEAM1': 'Team 1',
    'NCZONE_MATCH_TEAM2': 'Team 2',
    'NCZONE_MATCH_VS': 'VS',
    'NCZONE_MATCH_HAVE_BET': 'Tipps',
    'NCZONE_MATCH_NO_BETS': 'Keine Tipps',

    'NCZONE_LOADING': 'Lade Daten...',
    'NCZONE_ERROR_LOADING': 'Fehler beim Laden der Daten',
    'NCZONE_NO_PMATCHES': 'Keine vergangenen Spiele',
    'NCZONE_NO_RMATCHES': 'Keine laufenden Spiele',
    'NCZONE_NO_ACTIVE_PLAYERS': 'Keine aktiven Spieler',

    'NCZONE_TABLE_HEADER_NAME': 'Name',
    'NCZONE_TABLE_HEADER_GAMES': 'G',
    'NCZONE_TABLE_HEADER_WINS': 'W',
    'NCZONE_TABLE_HEADER_LOSSES': 'L',
    'NCZONE_TABLE_HEADER_TIES': 'T',
    'NCZONE_TABLE_HEADER_WINRATE': 'Winrate',
    'NCZONE_TABLE_HEADER_STREAK': 'S',
    'NCZONE_TABLE_HEADER_RATING_CHANGE': 'R+-',
    'NCZONE_TABLE_HEADER_RATING': 'R',
    'NCZONE_TABLE_HEADER_STREAK_RATING': 'SR',
    'NCZONE_TABLE_HEADER_ACTIVITY': 'Aktivität',
    'NCZONE_TABLE_FOOTER_AVERAGE': 'Durchschnitt',

    'NCZONE_DRAW_PREVIEW': 'Losevorschau',
    'NCZONE_DRAW_DO_YOU_WANT_TO_DRAW': 'Möchtest du losen?',
    'NCZONE_DRAW_LOGGED_IN_PLAYERS': 'Folgende Spieler sind eingeloggt:',
    'NCZONE_DRAW_PREVIEW_NO_PLAYERS': 'Es sind nicht genügend Spieler zum Losen da.',

    'NCZONE_REPLACE_PREVIEW_NO_PLAYERS': 'Es ist kein Spieler zum Austauschen da.',
    'NCZONE_REPLACE_PLAYER': 'Willst du ',
    'NCZONE_REPLACE_BY_PLAYER': ' durch ',
    'NCZONE_REPLACE_DO_YOU_WANT_TO_REPLACE': ' ersetzen?',

    'NCZONE_REASON_NOT_ALLOWED_TO_LOGIN': 'Nicht erlaubt einzuloggen.',
    'NCZONE_REASON_NOT_ALLOWED_TO_DRAW': 'Nicht erlaubt zu losen.',
    'NCZONE_REASON_NOT_ALLOWED_TO_POST_RESULT': 'Nicht erlaubt Ergebnis zu posten.',
    'NCZONE_REASON_NOT_ALLOWED_TO_POST_OTHER_RESULT': 'Nicht erlaubt das Ergebnis von jemand anderem zu posten.',
    'NCZONE_REASON_NOT_AN_ACTIVATED_PLAYER': 'Kein aktivierter Spieler.'
  },
  'en': {
    // civs
    'NCZONE_AZTECS': 'Aztecs',
    'NCZONE_BERBERS': 'Berbers',
    'NCZONE_BRITONS': 'Britons',
    'NCZONE_BURMESE': 'Burmese',
    'NCZONE_BYZANTINES': 'Byzantines',
    'NCZONE_CELTS': 'Celts',
    'NCZONE_CHINESE': 'Chinese',
    'NCZONE_ETHOPIANS': 'Ethopians',
    'NCZONE_FRANKS': 'Franks',
    'NCZONE_GOTHS': 'Goths',
    'NCZONE_HUNS': 'Huns',
    'NCZONE_INCAS': 'Incas',
    'NCZONE_INDIANS': 'Indians',
    'NCZONE_ITALIANS': 'Italians',
    'NCZONE_JAPANESE': 'Japanese',
    'NCZONE_KHMER': 'Khmer',
    'NCZONE_KOREANS': 'Koreans',
    'NCZONE_MAGYARS': 'Magyars',
    'NCZONE_MALAY': 'Malay',
    'NCZONE_MALIANS': 'Malians',
    'NCZONE_MAYANS': 'Mayans',
    'NCZONE_MONGOLS': 'Mongols',
    'NCZONE_PERSIANS': 'Persians',
    'NCZONE_PORTUGUESE': 'Portuguese',
    'NCZONE_SARACENS': 'Saracens',
    'NCZONE_SLAVS': 'Slavs',
    'NCZONE_SPANISH': 'Spanish',
    'NCZONE_TEUTONS': 'Teutons',
    'NCZONE_TURKS': 'Turks',
    'NCZONE_VIETNAMESE': 'Vietnamese',
    'NCZONE_VIKINGS': 'Vikings',

    'NCZONE_MATCH_RESULT': 'Result',
    'NCZONE_MATCH_POST_RESULT': 'Choose a result',
    'NCZONE_MATCH_POST_WIN_TEAM1': 'Team 1',
    'NCZONE_MATCH_POST_WIN_TEAM2': 'Team 2',
    'NCZONE_MATCH_POST_OW': 'No result',
    'NCZONE_MATCH_SEND_RESULT': 'Send result',

    'NCZONE': 'nC Zone',
    'NCZONE_OK': 'OK',
    'NCZONE_CANCEL': 'Cancel',

    'NCZONE_LOGGEDIN': 'Logged in',
    'NCZONE_LOGIN': 'Login',
    'NCZONE_LOGOUT': 'Logout',
    'NCZONE_DRAW': 'Draw',
    'NCZONE_NO_LOGIN': 'No one logged in :(',

    'NCZONE_LINKS': 'Links',
    'NCZONE_ZONE': 'Teh Zone',
    'NCZONE_INFORMATION': 'Max important information',

    'NCZONE_RMATCHES': 'Running matches',
    'NCZONE_PMATCHES': 'Past matches',
    'NCZONE_PLAYERS_TABLE': 'Player list',
    'NCZONE_MENU_INFORMATION': 'Information',
    'NCZONE_RULES': 'Rules',
    'NCZONE_SETTINGS': 'Settings',
    'NCZONE_SWITCH_LANG': 'Auf Deutsch wechseln',

    'NCZONE_MATCH_MATCH': 'Match',
    'NCZONE_MATCH_DRAWER': 'Drawed by',
    'NCZONE_MATCH_RESULT_POSTER': 'Posted by',
    'NCZONE_MATCH_WINNER': 'Winner',
    'NCZONE_MATCH_WINNER_NO_RESULT': 'No result',
    'NCZONE_MATCH_TEAM': 'Team',
    'NCZONE_MATCH_MAP': 'Map',
    'NCZONE_MATCH_CIVS': 'Civilizations',
    'NCZONE_MATCH_CIVS_BOTH': 'Both teams',
    'NCZONE_MATCH_CIVS_TEAM1': 'Team 1',
    'NCZONE_MATCH_CIVS_TEAM2': 'Team 2',
    'NCZONE_MATCH_START_TIME': 'Start time',
    'NCZONE_MATCH_TIME_SINCE_DRAW': 'Time since draw',
    'NCZONE_MATCH_LENGTH': 'Length',
    'NCZONE_MATCH_TEAMS': 'Teams',
    'NCZONE_MATCH_TEAM1': 'Team 1',
    'NCZONE_MATCH_TEAM2': 'Team 2',
    'NCZONE_MATCH_VS': 'VS',
    'NCZONE_MATCH_HAVE_BET': 'Bets',
    'NCZONE_MATCH_NO_BETS': 'No bets',

    'NCZONE_LOADING': 'Loading...',
    'NCZONE_ERROR_LOADING': 'Error loading data',
    'NCZONE_NO_PMATCHES': 'No past matches',
    'NCZONE_NO_RMATCHES': 'No running matches',
    'NCZONE_NO_ACTIVE_PLAYERS': 'No active players',

    'NCZONE_TABLE_HEADER_NAME': 'Name',
    'NCZONE_TABLE_HEADER_GAMES': 'G',
    'NCZONE_TABLE_HEADER_WINS': 'W',
    'NCZONE_TABLE_HEADER_LOSSES': 'L',
    'NCZONE_TABLE_HEADER_TIES': 'T',
    'NCZONE_TABLE_HEADER_WINRATE': 'Winrate',
    'NCZONE_TABLE_HEADER_STREAK': 'S',
    'NCZONE_TABLE_HEADER_RATING_CHANGE': 'R+-',
    'NCZONE_TABLE_HEADER_RATING': 'R',
    'NCZONE_TABLE_HEADER_STREAK_RATING': 'SR',
    'NCZONE_TABLE_HEADER_ACTIVITY': 'Activity',
    'NCZONE_TABLE_FOOTER_AVERAGE': 'Average',

    'NCZONE_DRAW_PREVIEW': 'Draw preview',
    'NCZONE_DRAW_DO_YOU_WANT_TO_DRAW': 'Do you want to draw?',
    'NCZONE_DRAW_LOGGED_IN_PLAYERS': 'The following players are logged in:',
    'NCZONE_DRAW_PREVIEW_NO_PLAYERS': 'There are not enough players to draw.',

    'NCZONE_REASON_NOT_ALLOWED_TO_LOGIN': 'Not allowed to login.',
    'NCZONE_REASON_NOT_ALLOWED_TO_DRAW': 'Not allowed to draw.',
    'NCZONE_REASON_NOT_ALLOWED_TO_POST_RESULT': 'Not allowed to post match results.',
    'NCZONE_REASON_NOT_ALLOWED_TO_POST_OTHER_RESULT': 'Not allowed to post someone else\'s match results.',
    'NCZONE_REASON_NOT_AN_ACTIVATED_PLAYER': 'Not an activated player.'
  }
}
