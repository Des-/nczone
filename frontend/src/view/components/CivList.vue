<template>
    <span>
        <span v-for="(civ, idx) in list" :key="idx">
            <span class="zone-match-civ">{{ $t(civ.title) }}<span class="zone-match-civ-tooltip">{{ civ.multiplier }}</span></span>
            <span v-if="idx+1 < list.length">, </span>
        </span>
    </span>
</template>
<script>
export default {
  name: 'nczone-civ-list',
  props: {
    list: {
      type: Array,
      required: true
    }
  }
}
</script>
