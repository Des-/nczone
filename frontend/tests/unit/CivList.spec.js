import { shallowMount, createLocalVue } from '@vue/test-utils'
import CivList from '@/view/components/CivList'
import VueI18n from 'vue-i18n'
import lang from '@/lang'

describe('CivList.vue', () => {
  const localVue = createLocalVue()
  localVue.use(VueI18n)

  const i18n = new VueI18n({
    locale: 'de',
    messages: lang
  })

  const list = [
    { title: 'NCZONE_AZTECS', multiplier: 2 },
    { title: 'NCZONE_BRITONS', multiplier: 1 }
  ]
  const cmp = shallowMount(CivList, {
    localVue: localVue,
    i18n,
    propsData: { list }
  })

  test('Should match snapshot', () => {
    expect(cmp.html()).toMatchSnapshot()
  })
})
