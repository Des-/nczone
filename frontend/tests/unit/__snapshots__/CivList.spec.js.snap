// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivList.vue Should match snapshot 1`] = `
<span><span><span class="zone-match-civ">Azteken<span class="zone-match-civ-tooltip">2</span></span> <span>, </span></span><span><span class="zone-match-civ">Briten<span class="zone-match-civ-tooltip">1</span></span>
<!----></span></span>
`;
